Fix #59 add encryption key config option

fangpenlin opened the pull request:
<!-- Please check the completed items below -->
### PR checklist

- [x] An issue/feature request has been created for this PR
- [x] Pull Request title clearly describes the work in the pull request and the Pull Request description provides details about how to validate the work. Missing information here may result in a delayed response.
- [x] File the PR against the `main` branch
- [x] The code in this PR is covered by unit tests

#### Link to issue/feature request: *#59*

#### Description
Provide alternative option to pass in `PublicKey` instead of `Certificate` for request payload encryption.

danny-gallagher commented:
Thanks @fangpenlin for creating the PR!

After some testing it looks like the encryption key fingerprint is no longer generated when the public encryption key is used.
We'll need to update the [computeEncryptionKeyFingerprintWhenNeeded](https://github.com/Mastercard/client-encryption-java/blob/main/src/main/java/com/mastercard/developer/encryption/EncryptionConfigBuilder.java#L24) function to account for the new key type.

fangpenlin commented:
Hi @danny-gallagher, thank you so much for pointing out the issue.

I have addressed it and also added test cases for that. Could you please review it again? Thanks :pray:

danny-gallagher reviewed (approved):
LGTM 🚀
